File: Programmers/3진법 뒤집기.py
# 몫,나머지 = divmod(n,값)
# int(숫자,n진수로 바꿀 변수) 로 하면 변환됨

# 테스트케이스
n=45

def solution(n):
    answer = 0
    number=''

    while n>0:
        q=n//3
        r=n%3
        number=number+str(r)
        n=q


    return int(number,3)

print(solution(n))

File: Programmers/3진법 뒤집기.cpp
#include <bits/stdc++.h>


using namespace std;

int solution(int n) {
    int answer = 0;

    string str;

    while(true){
        str += to_string(n%3);

        if(n<3)
            break;

        n = n/3;

    }

    for(int i=0; i<str.size(); i++){
        answer += pow(3,str.size()-i-1) * (str[i]-'0');
    }

    return answer;
}
